lighthouse-core/scripts/publish-prepare-commit.sh

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env bash
+
+TXT_BOLD=$(tput bold)
+TXT_DIM=$(tput setaf 245)
+TXT_RESET=$(tput sgr0)
+
+DIRNAME="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+LH_ROOT="$DIRNAME/../.."
+cd "$LH_ROOT"
+
+set -euxo pipefail
+
+if [[ "$#" -ne 1 ]]; then
+  echo "You must specify the version to prepare a commit for!"
+  exit 1
+fi
+
+OLD_VERSION=$(node -e "console.log(require('./package.json').version)")
+NEW_VERSION=$1
+BRANCH_NAME="bump_$NEW_VERSION"
+SEMVER_PATTERN="[0-9]*\.[0-9]*\.[0-9]*"
+
+if [[ $(echo "$NEW_VERSION" | sed 's/[0-9]*\.[0-9]*\.[0-9]*/SECRET_REPLACE/g') != "SECRET_REPLACE" ]]; then
+ echo "Incorrect version format. Must be x.x.x"
+ exit 1
+fi
+
+if [[ -n "$(git status --porcelain)" ]]; then
+  echo "Repo has changes to the files! Commit or stash the changes to continue."
+  exit 1
+fi
+
+# Checkout a new branch for the version commit
+git fetch origin master
+git checkout origin/master
+git branch -D "$BRANCH_NAME" || true
+git checkout -b "$BRANCH_NAME"
+
+# Install the dependencies.
+yarn install
+
+# Bump the version in package.json and clients/extension/manifest.json
+NEEDLE="^  \"version\": \"$SEMVER_PATTERN\""
+REPLACEMENT="  \"version\": \"$NEW_VERSION\""
+
+sed -i '' "s/$NEEDLE/$REPLACEMENT/g" package.json clients/extension/manifest.json
```

Comment:
nice. handy that you can do both in the same cmd.

do we need an `|| exit 1` on the sed for failure states? dunno if it'd have one.

Reply:
we've got `set -euxo pipefail` so any failing command should fail

or is that not what you're referring to?